src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -123,6 +124,15 @@ internal bool ShouldMonitorHostHealth
 
         public async Task StartAsync(CancellationToken cancellationToken)
         {
+            // Shutdown if RunFromZipFailed
+            string runFromPackageFailedPath = Path.Combine(_applicationHostOptions.CurrentValue.ScriptPath, ScriptConstants.RunFromPackageFailedFileName);
+            if (File.Exists(runFromPackageFailedPath))
```

Comment:
Use Async file API

Reply:
There was no ExistsAsync so I added something to FileUtility like other methods in there

Reply:
Actually, this was causing some unit tests to fail so I reverted to using File.Exists.